packages/core/components/EditMetadata/index.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import ExistingAnnotationPathway from "./ExistingAnnotationPathway";
+import NewAnnotationPathway from "./NewAnnotationPathway";
+import ChoiceGroup from "../ChoiceGroup";
+import { TOP_LEVEL_FILE_ANNOTATION_NAMES } from "../../constants";
+import { interaction, metadata, selection } from "../../state";
+
+import styles from "./EditMetadata.module.css";
+
+enum EditMetadataPathway {
+    EXISTING = "existing",
+    NEW = "new",
+}
+
+/**
+ * Form that acts as a wrapper for both metadata editing pathways (new vs existing annotations).
+ * Performs all necessary state selections on render and passes data as props to child components
+ */
+export default function EditMetadataForm() {
+    const dispatch = useDispatch();
+    const onDismiss = () => {
+        dispatch(interaction.actions.hideVisibleModal());
+    };
+    const fileSelection = useSelector(selection.selectors.getFileSelection);
+    const fileCount = fileSelection.count();
+    // Don't allow users to edit top level annotations (e.g., File Name)
+    const annotationOptions = useSelector(metadata.selectors.getSortedAnnotations)
```

Comment:
Unsure why I have never attached JS array functions onto the selector like this... looks clean!